pkg/scheduler/cache/nodeinfo.go

```diff
@@ -129,3 +217,59 @@ func (n *NodeInfo) RemovePod(key string, pod *v1.Pod) {
 	n.QoSResourcesNonZeroRequested.ReclaimedMilliCPU -= podInfo.QoSResourcesNonZeroRequested.ReclaimedMilliCPU
 	n.QoSResourcesNonZeroRequested.ReclaimedMemory -= podInfo.QoSResourcesNonZeroRequested.ReclaimedMemory
 }
+
+func (n *NodeInfo) AddAssumedPod(pod *v1.Pod) {
+	n.Mutex.Lock()
+	defer n.Mutex.Unlock()
+	n.AssumedPodResources.AddPod(pod)
+}
+
+func (n *NodeInfo) DeleteAssumedPod(pod *v1.Pod) {
+	n.Mutex.Lock()
+	defer n.Mutex.Unlock()
+
+	n.AssumedPodResources.DeletePod(pod)
+}
+
+func (n *NodeInfo) GetResourceTopologyCopy(filterFn podFilter) *ResourceTopology {
+	n.Mutex.RLock()
+	defer n.Mutex.RUnlock()
+
+	if n.ResourceTopology == nil {
+		return nil
+	}
+
+	return n.ResourceTopology.WithPodReousrce(n.AssumedPodResources, filterFn)
+}
+
+func (n *NodeInfo) GetAssumedPodAffinityInfo() AssumedPodAffinityInfo {
+	n.Mutex.RLock()
+	defer n.Mutex.RUnlock()
+	return n.AssumedPodAffinityInfo
```

Comment:
AssumedPodAffinityInfo is not thread safe, maybe it is better to return a deepcopy ?